engine/src/main/java/org/hibernate/search/backend/LuceneWork.java

```diff
@@ -61,7 +59,14 @@ public String getIdInString() {
 		return idInString;
 	}
 
-	public abstract <T> T getWorkDelegate(WorkVisitor<T> visitor);
```

Comment:
Note that this removes a previously publicly exposed method. IMO that's acceptable though, given that the parameter type resided in "impl", making that previous signature kind of "faulty" anyways.